src/core/reference/include/openvino/reference/scatter_elements_update.hpp

```diff
@@ -247,5 +217,86 @@ void scatter_elem_update_with_reduction(const DataType* input_data,
         }
     }
 }
+
+template <
+    typename DataType,
+    typename IndicesType,
+    typename std::enable_if<std::is_same<typename std::decay<IndicesType>::type, int64_t>::value>::type* = nullptr>
+void scatter_elem_update(const DataType* input_data,
+                         const IndicesType* indices,
+                         const DataType* updates,
+                         const int64_t axis,
+                         DataType* out_buf,
+                         const Shape& data_shape,
+                         const Shape& indices_shape,
+                         const Reduction reduction_type = Reduction::NONE,
+                         const bool use_init_val = true) {
+    // Copy inputs to out
+    std::memcpy(out_buf, input_data, sizeof(DataType) * shape_size(data_shape));
+
+    if (reduction_type != Reduction::NONE) {
+        scatter_elem_update_with_reduction(indices,
+                                           updates,
+                                           axis,
+                                           out_buf,
+                                           data_shape,
+                                           indices_shape,
+                                           reduction_type,
+                                           use_init_val);
+    } else {
+        scatter_elem_update_no_reduction(sizeof(DataType),
+                                         indices,
+                                         reinterpret_cast<const char*>(updates),
+                                         axis,
+                                         reinterpret_cast<char*>(out_buf),
+                                         data_shape,
+                                         indices_shape,
+                                         reduction_type,
+                                         use_init_val);
+    }
+}
+
+template <
+    typename DataType,
+    typename IndicesType,
+    typename std::enable_if<!std::is_same<typename std::decay<IndicesType>::type, int64_t>::value>::type* = nullptr>
+void scatter_elem_update(const DataType* input_data,
+                         const IndicesType* indices,
+                         const DataType* updates,
+                         const int64_t axis,
+                         DataType* out_buf,
+                         const Shape& data_shape,
+                         const Shape& indices_shape,
+                         const Reduction reduction_type = Reduction::NONE,
+                         const bool use_init_val = true) {
+    // Copy inputs to out
+    std::memcpy(out_buf, input_data, sizeof(DataType) * shape_size(data_shape));
+
+    const auto indices_count = shape_size(indices_shape);
```

Comment:
Instead make copy of function can block related indices be specialized (as helper function)?